nodejs/eks/cluster.ts

```diff
@@ -216,9 +221,13 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
     } else if (args.instanceRole) {
         // Create an instance profile if using a default node group
         if (!args.skipDefaultNodeGroup) {
-            instanceProfile = new aws.iam.InstanceProfile(`${name}-instanceProfile`, {
-                role: args.instanceRole,
-            }, { parent: parent });
+            if (args.instanceProfileName) {
+              instanceProfile = aws.iam.InstanceProfile.get(args.instanceProfileName, args.instanceProfileName);
```

Comment:
I would use `${name}-instanceProfile` for the name here to allow `args.instanceProfileName` to be a `pulumi.Input<string>`. i.e.

```suggestion
              instanceProfile = aws.iam.InstanceProfile.get(`${name}-instanceProfile`, args.instanceProfileName, undefined, { parent: parent });
```